lib/msf/core/post/file.rb

```diff
@@ -322,6 +325,55 @@ def exist?(path)
 
   alias exists? exist?
 
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  #
+  def check_win_path_permissions(path, perms)
+    # Grab permissions of the path in question and extract relevant info using a regex.
+    # Also strip extra whitespace before comparison occurs.
+    permissions_file_raw = cmd_exec("icacls \"#{path}\"").to_s
+    permissions_file_array = permissions_file_raw.scan(/^.+? (.+?\\.+?):(\(.*\))/)
+    permissions_file_array_length = permissions_file_array.length
+    for i in 0...permissions_file_array_length
+      permissions_file_array[i][0].strip!
+      permissions_file_array[i][1].strip!
+    end
+
+    # Grab info on the groups the current user belongs to so that we can do our comparisons.
+    cuser_groups = cmd_exec("whoami /groups").to_s
+    cuser_groups_array = cuser_groups.scan(/^(?:[\w]+ ?[\w\-]*+ ?[\w\-]*+\\?[\w\-]*+ ?[\w\-]*+ ?[\w\-]*+ ?)/)
```

Comment:
At the very least a comment with a sample for that this is parsing would be helpful.  It may be of value to have a method call that parse the line in parts vs a regex here.

Reply:
Unfortunately as this is only a single group capture and I had a tough time making this regex work, I wasn't able to simplify things. That being said I did move this into its own function with https://github.com/rapid7/metasploit-framework/pull/15987/commits/f3f907b3f630547b2d4aef4296545ee33e1a8fe1 and added comments for a sample of this and `icalcs` output so that its easier to see what we are matching on.

Let me know if these changes make sense 👍